FSF-3221 Integration Tests

andrelds11 opened the pull request.

marhcouto reviewed (changes requested):
Sorry for taking so long. Overall ok but tests need further documentation. Moreover, I don't think these were very thought through, we can discuss this tomorrow further

andrelds11 commented on master/include/comm/communicator.hpp:
```diff
@@ -105,18 +107,17 @@ inline Communicator::Communicator() {
     double right_wheel_rpm = (buf[4] << 24) | (buf[3] << 16) | (buf[2] << 8) | buf[1];
     right_wheel_rpm *= WHEEL_PRECISION; // convert back adding decimal part
-    _systemData->sensors._rl_wheel_rpm = right_wheel_rpm;
+    _systemData->sensors._right_wheel_rpm = right_wheel_rpm;
```
consistency

marhcouto reviewed (changes requested):
We should talk about most of these comments. Main topics:
- the unit tests are giving me headaches
- a bit better documentation on the integration tests
- the errors mentioned
- not using function (to_ready) every time you can

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+        as_state.calculateState();
+    }
+    
+    sd.digitalData.watchdog_state = false;
```
Watchdog needs to go low in the initial checkup sequence

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+
+/**
+ * @brief Test function to validate AS_OFF to AS_Ready transition 
+*/
```
Off to ready success transition with no other conditions or restraints

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+/**
+ * @brief Test function to validate AS_READY doesn't revert to AS_Ready 
+*/
+void test_off_to_ready_wayback() {
```
It doesn't. The test is to validate it doesn't happen anymore. Changed the name slightly to make it clearer

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+    }
+
+    uint8_t msg[8] = {0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+    communicator.resStateCallback(msg);
```
Communicator is always used when possible for double validation (redundancy), maybe just with the exception of timestamps. Digital Data is always changed directly to do evaluate it with no breadboard inputs

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+        sd.digitalData.watchdogTimestamp.reset();
+        as_state.calculateState();
+    }
+    TEST_ASSERT_EQUAL(State::AS_DRIVING, as_state.state); // still within threshold, okay
```
Unit tests for the checkupmanager specific transitions are in the checkupmanager tests already

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+    reset();
+    TEST_ASSERT_EQUAL(State::AS_OFF, as_state.state);
+    uint8_t hydraulic_msg[] = {HYDRAULIC_LINE, 0xf8, 0x00};
+    communicator.c1Callback(hydraulic_msg);
```
Included in the initial checkup. Necessary for it to be ready. Will add a comment on that

andrelds11 commented on master/test/test_integration/test_integration.cpp:
```diff
@@ -0,0 +1,505 @@
+void test_flow_driving() {
+    reset();
+    TEST_ASSERT_EQUAL(State::AS_OFF, as_state.state);
+    uint8_t hydraulic_msg[] = {HYDRAULIC_LINE, 0xf8, 0x00};
```
Should I use decimals? What is the idea?

marhcouto commented:
I approved but the steering angle from steering's ID is wrong, it is 0x295D and it is an extended ID. Consult the PR in ROS-CAN if you need to check how it is read

andrelds11 commented:
> I approved but the steering angle from steering's ID is wrong, it is 0x295D and it is an extended ID. Consult the PR in ROS-CAN if you need to check how it is read

Sorry, I didn't notice this. I changed the id, from what I saw in the other rep, it is normally read so I just changed the id, can you confirm this?
